.github/workflows/upload-and-deploy.yaml

```diff
@@ -46,13 +47,162 @@ jobs:
         run: echo "NAMESPACE=$GITHUB_REF_NAME" >> $GITHUB_ENV
 
       - name: Copy files to templates bucket, use dev cloudformation bucket
-        run: python src/scripts/manage_artifacts/artifacts.py --upload --namespace $NAMESPACE --cfn_bucket ${{ vars.CFN_BUCKET }}
+        run: >
+          python src/scripts/manage_artifacts/artifacts.py
+          --upload
+          --namespace $NAMESPACE
+          --cfn_bucket ${{ vars.CFN_BUCKET }}
+
+      - name: Copies over test files from ingestion bucket
+        run: >
+          aws s3 sync s3://recover-dev-ingestion/ s3://$DEV_INPUT_DATA_BUCKET/$NAMESPACE/
+          --exclude "owner.txt"
+          --exclude "test_upload.rtf"
+
+
+  nonglue-unit-tests:
+    name: Runs unit tests that are not dependent on aws-glue package resources
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    environment: develop
+    steps:
+      - name: Setup code, pipenv, aws
+        uses: Sage-Bionetworks/action-pipenv-aws-setup@v3
+        with:
+          role_to_assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
+          role_session_name: ${{ github.event.repository.name }}-${{ github.run_id }}-nonglue-unit-tests
+          python_version: ${{ env.PYTHON_VERSION }}
+
+      - name: Test lambda scripts with pytest
+        run: |
+          pipenv run python -m pytest tests/test_s3_event_config_lambda.py -v
+          pipenv run python -m pytest tests/test_s3_to_glue_lambda.py -v
+
+      - name: Test synapse folders for STS access with pytest
+        run: >
+          pipenv run python -m pytest tests/test_setup_external_storage.py
+          --test-synapse-folder-id syn51084525
+          --test-ssm-parameter synapse-recover-auth
+          -v
+
+
+  pytest-docker:
+    # A ECR repo called pytest would need to exist in the AWS account
+    # we are pushing docker images to prior to this
+    name: Build and push testing docker images to the pytest ECR repository.
+    needs: pre-commit
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      matrix:
+        include:
+          - tag_name: aws_glue_3
+            dockerfile: tests/Dockerfile.aws_glue_3
+          - tag_name: aws_glue_4
+            dockerfile: tests/Dockerfile.aws_glue_4
+    environment: develop
+    steps:
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
+          aws-region: "us-east-1"
+          # unmasking of the AWS account ID allows the acct id to pass through outputs
+          mask-aws-account-id: "no"
```

Comment:
Why do we need to pass the account ID through outputs?

Reply:
So BD's account ID is also unmasked.
Essentially because we need to pass `ecr-registry` from the job `pytest-docker` whose value is `steps.login-ecr.outputs.registry` which contains the account ID, through so `glue-unit-tests` job can use it to get the docker container path

Reply:
Yeah the output will be a bit different from BD's just because we use the `matrix` strategy so I had to tweak it a bit.